python/flow_sdk/shim_singer_sdk.py

```diff
@@ -92,14 +168,36 @@ def discover(self, discover: request.Discover) -> response.Discovered:
                     for p in entry.key_properties
                 ]
             else:
+                logger.info(f"Stream {stream.name} does not have a primary key, using '/_meta/row_id' as the key.") 
+                if entry.replication_method == REPLICATION_INCREMENTAL:
+                    logger.warn(f"Stream {stream.name} is using {REPLICATION_INCREMENTAL} replication without a primary key. If this is not an append-only data source, updates to existing documents will be captured as inserts with new keys. Switch to {REPLICATION_FULL_TABLE} to fix this.")
+                
                 key = ["/_meta/row_id"]
+                row_id_type = Property("row_id", IntegerType, required=True)
+
+                if entry.schema.properties is not None:
```

Comment:
Shouldn't we schematize `/_meta/row_id` in *all* cases? It's always there, right?